pkg/controller/swiftproxy/swiftproxy_controller.go

```diff
@@ -221,7 +258,41 @@ func (r *ReconcileSwiftProxy) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, r.updateStatus(swiftProxy, deployment)
+	return reconcile.Result{}, r.updateStatus(swiftProxy, deployment, svc)
+}
+
+func (r *ReconcileSwiftProxy) ensureLabelExists(sp *contrail.SwiftProxy) error {
+	if len(sp.Labels) != 0 {
+		return nil
+	}
+
+	sp.Labels = label.New(contrail.SwiftProxyInstanceType, sp.Name)
+	return r.client.Update(context.Background(), sp)
+}
+
+func (r *ReconcileSwiftProxy) ensureServiceExists(swiftProxy *contrail.SwiftProxy) (*core.Service, error) {
+	svc := &core.Service{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      swiftProxy.Name + "-swift-proxy",
+			Namespace: swiftProxy.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, svc, func() error {
+		listenPort := int32(swiftProxy.Spec.ServiceConfiguration.ListenPort)
+		nodePort := int32(0)
+		for i, p := range svc.Spec.Ports {
+			if p.Port == listenPort {
+				nodePort = svc.Spec.Ports[i].NodePort
+			}
+		}
+		svc.Spec.Ports = []core.ServicePort{
+			{Port: listenPort, Protocol: "TCP", NodePort: nodePort},
+		}
+		svc.Spec.Selector = swiftProxy.Labels
+		svc.Spec.Type = core.ServiceTypeLoadBalancer
```

Comment:
If we are exposing the service as "LoadBalancer" type, why are we using ClusterIP? Why not LoadBalancerIP?